### PR TITLE
Add a calendar view to time entries

### PR DESCRIPTION
**Pull request by Cronos87**
This calendar view allow to see the selected month total hours logged by days.

Can be used using the following commands:

Current month: `arcli log calendar`
Selecting a month: `arcli log calendar -m "2019-12"`

Here is how it looks like:

CleanShot 2021-11-08 at 17 33 45@2x

### CODE REVIEW

**Comment by mightymatth** [comment]
When I designed CLI commands, `arcli log` is meant to be used as an action to log time for an issue or a project, not to list something. I suggest you extract it to a new route called view (abbreviated v) so we can have something like:

```
arcli view calendar
arcli v calendar
arcli v cal
```

**Comment by Cronos87** [comment]
View as singular or plural?

File: cmd/view.go [hunk 1]
```diff
@@ -0,0 +1,276 @@
...
+	// Check if the month format is correct.
+	var re = regexp.MustCompile(`current|[\d]{4}-[\d]{2}`)
+
+	if !re.MatchString(spentOnMonth) {
+		return errors.New("the format is not correct. The following formats are supported: \"current\", \"2021-11\"")
+	}
+
+	fmt.Print("Loading data...\r")
+
+	// Define the days of th week.
```
Comment (mightymatth) [inline cmd/view.go:70]: This comment is not really needed in my opinion.

File: cmd/view.go [hunk 2]
```diff
@@ -0,0 +1,276 @@
...
+	var daysOfWeek = []string{"Sun", "Mon", "Tue", "Wed", "Thu", "Fri", "Sat"}
+
+	// Get the date depending on the user's choice.
```
Comment (mightymatth) [inline cmd/view.go:73]: Also, this comment is not really needed.

File: cmd/view.go [hunk 3]
```diff
@@ -0,0 +1,276 @@
...
+	var date *now.Now
+
+	if spentOnMonth == "current" {
+		date = now.With(time.Now())
+	} else {
+		var dateParsed, _ = time.Parse("2006-01-02", fmt.Sprintf("%s-01", spentOnMonth))
+		date = now.With(dateParsed)
+	}
+
+	// Get the wanted date.
```
Comment (mightymatth) [inline cmd/view.go:83]: Comment not needed.

File: cmd/view.go [hunk 4]
```diff
@@ -0,0 +1,276 @@
...
+	var formattedDate = date.Format("January 2006")
+
+	// Get the totals days in the month.
```
Comment (mightymatth) [inline cmd/view.go:86]: Comment not needed.

File: cmd/view.go [hunk 5]
```diff
@@ -0,0 +1,276 @@
...
+	var _, _, totalDaysInMonth = date.EndOfMonth().Date()
+
+	// Get the weekday of the first day of the month.
```
Comment (mightymatth) [inline cmd/view.go:89]: Comment not needed.

File: cmd/view.go [hunk 6]
```diff
@@ -0,0 +1,276 @@
...
+	var weekday = int(date.BeginningOfMonth().Weekday())
+
+	// Calculate the total weeks in the month.
```
Comment (mightymatth) [inline cmd/view.go:92]: Comment not needed.

File: cmd/view.go [hunk 7]
```diff
@@ -0,0 +1,276 @@
...
+	var totalWeeksInMonth int = int(math.Ceil(float64(totalDaysInMonth+weekday) / 7))
+
+	// Get the time entries for the month.
```
Comment (mightymatth) [inline cmd/view.go:95]: Comment not needed.

**Review by mightymatth — COMMENTED** [review]
Fix unnecessary comments.

**Comment by Cronos87** [comment]
I'll fix this after my working time.

**Review by mightymatth — COMMENTED** [review]
We are already using [goterm](https://github.com/buger/goterm) package for writing structured data on the screen. Have you tried to use it before writing your own logic for this calendar view? Did you experience some issues with that? 
If you could make it with this package, there would be significantly less drawing logic.

**Comment by mightymatth** [comment]
Closing due to inactivity

**Pull request closed without merging**